datastream-server-restli/src/test/java/com/linkedin/datastream/server/dms/TestDatastreamResources.java

```diff
@@ -296,24 +295,16 @@ public void testSimultaneousStopDatastreamRequests() {
     PathKeys pathKey = Mockito.mock(PathKeys.class);
     Mockito.when(pathKey.getAsString(DatastreamResources.KEY_NAME)).thenReturn(datastreamName);
 
-    // Stop datastream.
     Assert.assertEquals(resource1.get(datastreamName).getStatus(), DatastreamStatus.READY);
 
+    // Stop datastream.
     ExecutorService executor = Executors.newFixedThreadPool(2);
     executor.execute(() -> resource1.stop(pathKey, false));
-    executor.execute(() -> resource1.stop(pathKey, false));
-
-    try {
-      executor.awaitTermination(100, TimeUnit.MILLISECONDS);
-    } catch (InterruptedException ignored) {
-    } finally {
-      executor.shutdownNow();
-    }
+    executor.execute(() -> resource2.stop(pathKey, false));
 
     // Retrieve datastream and check that is in STOPPED state.
-    Datastream ds = resource2.get(datastreamName);
-    Assert.assertNotNull(ds);
-    Assert.assertEquals(ds.getStatus(), DatastreamStatus.STOPPED);
+    Assert.assertTrue(PollUtils.poll(() -> resource1.get(datastreamName).getStatus() == DatastreamStatus.STOPPED, 100, 1000));
```

Comment:
Do we not need to check for resource2 status ?

Reply:
nope, they are just resource endpoints, both of them would give the same output. We just have two of them in the test to mock the scenario of having multiple endpoints requesting the status change.